### PR TITLE
fix: Bulk sell buttons incorrectly appear in blacksmith

### PR DESCRIPTION
**Pull request by ryanzhoudev**

### CODE REVIEW

**Review by kristofbolyai — COMMENTED** [review]
I am not sure if this is the way to go, the check is rather lose.

**Comment by magicus** [comment]
I agree, this looks imprecise. Why is the `SELL_DIALOGUE_TITLE` match not enough? Do the TM and Blacksmith sell dialog really have the exact same title?

**Comment by ryanzhoudev** [comment]
image
image

They are the exact same

**Comment by magicus** [comment]
I see. I'd rather see a solution where we check for the hover text of the "+" (set amount) button. (After all, we are counting on that existing if we are to be able to press it).

File: common/src/main/java/com/wynntils/features/overlays/TradeMarketBulkSellFeature.java [hunk 1]
```diff
@@ -48,33 +48,28 @@ public class TradeMarketBulkSellFeature extends Feature {
...
+        buttonsAdded = false;
     }
 
     @SubscribeEvent
     public void onSellDialogueUpdated(ContainerSetSlotEvent e) {
         if (!(McUtils.mc().screen instanceof ContainerScreen cs)) return;
+        if (!ComponentUtils.getUnformatted(
+                        cs.getMenu().getSlot(AMOUNT_ITEM_SLOT).getItem().getHoverName())
+                .equals("Click to Set Amount")) return;
         if (!ComponentUtils.getUnformatted(cs.getTitle()).equals(SELL_DIALOGUE_TITLE)) return;
```
Comment (magicus) [inline common/src/main/java/com/wynntils/features/overlays/TradeMarketBulkSellFeature.java]: Can you please check the title first? I'm worried that you might end up in a completely different container screen which might not even have a slot 11, and that this check would crash.

**Review by magicus — APPROVED** [review]

**Pull request merged**